_posts/2018-07-11-cockpit-172.md

```diff
@@ -0,0 +1,74 @@
+---
+title: Cockpit 172
+author: pitti
+date: 2018-07-11
+tags: cockpit linux
+slug: cockpit-172
+category: release
+summary: Cockpit with on-demand PCP install, flow control, and Python 3 support
+comments: 'true'
+---
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/). We release regularly.
+Here are the release notes from version 172.
+
+### System: Offer installation of PCP
+
+The System page now shows an "Enable persistent metrics…" link if the
+[PCP](https://pcp.io/) package is not installed. This is similar to the
+on-demand installation of the NFS client packages in
+[Cockpit 169](https://cockpit-project.org/blog/cockpit-169.html).
+
+![Install PCP on demand](/images/pcp-install-on-demand.png)
+
+### Software Updates: Improve layout in mobile mode
+
+The Software Updates page fixed some incorrect spacing and layout in small
+browser windows and mobile browsers:
+
+![PackageKit mobile mode](/images/packagekit-mobile-optimization.png)
+
+
+### Remove ability to drop privileges from navigation bar
+
+Cockpit previously showed a "Locked" or "Unlocked" status in the navigation
+bar, reflecting whether or not the "Reuse my password for privileged tasks"
+check box was enabled at log in. Clicking on "Unlocked" would lock the
+interface and drop administrator capabilities.
+
+This feature was poorly supported, as most pages did not properly respond to
+this change. It got replaced by an indicator whether the user has
+administrative privileges ("Privileged") and is not clickable.
+
+The ability to start cockpit without escalating privileges remains on the login
+screen. Privileges can also still be dropped from the "Authentication" user
+menu.
+
+![Privilege status](/images/cockpit-drop-privileges.png)
+
+### API: Introduce flow control for all channels
+
+It is possible to use the Cockpit API to transfer large amounts of data, such
+as downloading an Operating System image from a remote machine. This now
+ensures that links in the route from the source to the target machine don't
+have to buffer a large, and potentially unbounded, amount of data when the
+subsequent link was slower than the previous one. This could previously result
+in cockpit-bridge using an inordinate amount of memory.
```

Comment:
the main reason for the change was actually to prevent updates to other bits of the cockpit UI from getting stuck behind the massive transfer buffer, so it might be reasonable to mention that as well.